### PR TITLE
Room settings: Upgraded `RoomSettingsGetter` service.

### PR DESCRIPTION
**Pull request by KH-Amir-TN**
+ Added access codes handling to the service.
	+ Added `only_enabled` mode to fetch only enabled options.

## Testing Steps

> 1. Pull the code.
> 2. Install the dependencies `bundle install && npm|yarn install`.
> 3. Clean the previous assets build by running `rm app/assets/builds/*` (This won't remove .keep since it's hidden).
> 4. Clean the database and tmp files for a better isolation by running `rails tmp:clear && rails db:schema:cache:clear && rails db:drop && rails db:create && rails db:migrate:with_data`
> 5. Run the linter and specs `bundle exec rubocop --parallel && bundle exec rspec && npx eslint app/javascript/* --ext .jsx,.js`
> 6. Run `./bin/dev` to run the assets builders processes and the Puma server all at once.

### CODE REVIEW

**Comment by farhatahmad** [comment]
I'm a little confused - what's the purpose of this PR?

**Comment by KH-Amir-TN** [comment]
> I'm a little confused - what's the purpose of this PR?

Upgrading the `RoomSettingsGetter` service and the `room_settings_controller#show` API to:
1. Fetch access codes handling their forced enabled and disabled cases.
2. Be able to filter out forced disabled settings.

**Comment by farhatahmad** [comment]
What's the plan to distinguish `force_enabled`? I think the service is starting to become over complicated. Can we move some of this logic to the controller instead?

File: app/services/room_settings_getter.rb [hunk 1]
```diff
@@ -23,7 +24,14 @@ def call
...
     room_settings = room_settings.pluck(:name, :value).to_h
 
+    access_codes = room_settings.slice('glViewerAccessCode', 'glModeratorAccessCode')
+
     room_settings.merge!(@rooms_configs) # Merging rooms settings with its none optional configurations prioritizing forced configs over settings.
+    if @only_enabled
+      # Only enabled(optional|force enabled) setting values will be returned.
+      disabled_settings = @rooms_configs.filter { |_k, v| v == 'false' }
+      room_settings.except!(*disabled_settings.keys)
+    end
```
Comment (farhatahmad) [inline app/services/room_settings_getter.rb:34]: Let's move this to a private method `filter_disabled if @only_enabled`

File: app/services/room_settings_getter.rb [hunk 2]
```diff
@@ -23,7 +24,14 @@ def call
...
     room_settings_special_options = SPECIAL_OPTIONS.slice(*room_settings.keys) # Extracting the room settings special options to minimize iterations.
```
Comment (farhatahmad) [inline app/services/room_settings_getter.rb:35]: I would move the special options to its own private method as well

File: app/services/room_settings_getter.rb [hunk 3]
```diff
@@ -23,7 +24,25 @@ def call
...
+  end
+
+  private
+
+  def filter_disabled(room_settings:)
+    disabled_settings = @rooms_configs.filter { |_k, v| v == 'false' }
+    room_settings.except!(*disabled_settings.keys)
+  end
+
+  def infer_sepecials(room_settings:)
```
Comment (farhatahmad) [inline app/services/room_settings_getter.rb:45]: `specials` not `sepecials`

Comment (KH-Amir-TN) [reply app/services/room_settings_getter.rb:45]: 😅 😅 😅

**Review by farhatahmad — APPROVED** [review]

**Comment by farhatahmad** [comment]
@KH-Amir-TN Rubocop

**Pull request merged**